Examples/Algorithms/TrackFinding/src/TrackParamsLookupEstimation.cpp

```diff
@@ -0,0 +1,101 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "ActsExamples/TrackFinding/TrackParamsLookupEstimation.hpp"
+
+#include "Acts/EventData/SourceLink.hpp"
+#include "Acts/Surfaces/RectangleBounds.hpp"
+
+ActsExamples::TrackParamsLookupEstimation::TrackParamsLookupEstimation(
+    const Config& config, Acts::Logging::Level level)
+    : IAlgorithm("TrackParamsLookupEstimation", level),
+      m_cfg(std::move(config)) {
+  // Iterate over the reference layers and create
+  // track parameter accumulators
+  for (const auto& [geoId, refSurface] : m_cfg.refLayers) {
```

Comment:
Could you check the surface type in the constructor as well?

Reply:
Done